### DISCUSSION TITLE
ffmpeg broken

### DISCUSSION THREAD

**Discussion by akaase** [Homebrew/discussions#4855] [Everyday usage] [answered]
### Output of `brew config`

```shell
aaron@Aarons-MacBook-Air:~$ brew config
HOMEBREW_VERSION: 4.1.16
ORIGIN: https://github.com/Homebrew/brew
HEAD: c0c8a4dd4dd19ec16deb429abd50cdf82ea33b27
Last commit: 9 days ago
Core tap JSON: 23 Oct 08:08 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: /usr/bin/nano
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
aaron@Aarons-MacBook-Air:~$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  six
aaron@Aarons-MacBook-Air:~$ brew link six
Linking /opt/homebrew/Cellar/six/1.16.0_4... 
Error: Could not symlink lib/python3.10/site-packages/six.py
Target /opt/homebrew/lib/python3.10/site-packages/six.py
already exists. You may want to remove it:
  rm '/opt/homebrew/lib/python3.10/site-packages/six.py'

To force the link and overwrite all conflicting files:
  brew link --overwrite six

To list all files that would be deleted:
  brew link --overwrite --dry-run six
```

### Description of issue

Attempting to install and run ffmpeg using the following sequence and get the following error:

aaron@Aarons-MacBook-Air:~$ brew install ffmpeg
==> Downloading https://formulae.brew.sh/api/formula.jws.json

==> Downloading https://formulae.brew.sh/api/cask.jws.json

==> Downloading https://ghcr.io/v2/homebrew/core/ffmpeg/manifests/6.0_1-2
Already downloaded: /Users/aaron/Library/Caches/Homebrew/downloads/a839e675f8ecc4a6c3a689dae26e0deed55876713b87948b57ae6840dd6d5819--ffmpeg-6.0_1-2.bottle_manifest.json
==> Fetching ffmpeg
==> Downloading https://ghcr.io/v2/homebrew/core/ffmpeg/blobs/sha256:42d52253402650e0213379f160c55d84b65dd00f6399f164df05f114c74ebcc4
Already downloaded: /Users/aaron/Library/Caches/Homebrew/downloads/674b8adba245ad3d81c5130afc3c14372086dc35d08d705ebe1e90a771b5a4e5--ffmpeg--6.0_1.arm64_sonoma.bottle.2.tar.gz
==> Pouring ffmpeg--6.0_1.arm64_sonoma.bottle.2.tar.gz
🍺  /opt/homebrew/Cellar/ffmpeg/6.0_1: 284 files, 48.7MB
==> Running `brew cleanup ffmpeg`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
aaron@Aarons-MacBook-Air:~$ ffmpeg
dyld[17615]: Library not loaded: /opt/homebrew/opt/libunistring/lib/libunistring.2.dylib
  Referenced from: <8C2E9A39-E47F-3DB6-91DC-61930A86359B> /opt/homebrew/Cellar/libidn2/2.3.4/lib/libidn2.0.dylib
  Reason: tried: '/opt/homebrew/opt/libunistring/lib/libunistring.2.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libunistring/lib/libunistring.2.dylib' (no such file), '/opt/homebrew/opt/libunistring/lib/libunistring.2.dylib' (no such file), '/usr/local/lib/libunistring.2.dylib' (no such file), '/usr/lib/libunistring.2.dylib' (no such file, not in dyld cache), '/opt/homebrew/Cellar/libunistring/1.1/lib/libunistring.2.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/libunistring/1.1/lib/libunistring.2.dylib' (no such file), '/opt/homebrew/Cellar/libunistring/1.1/lib/libunistring.2.dylib' (no such file), '/usr/local/lib/libunistring.2.dylib' (no such file), '/usr/lib/libunistring.2.dylib' (no such file, not in dyld cache)Library not loaded: /opt/homebrew/opt/mbedtls/lib/libmbedcrypto.12.dylib
  Referenced from: <9AB6FD1A-FE12-309C-8F22-AE6259869D59> /opt/homebrew/Cellar/librist/0.2.7_1/lib/librist.4.dylib
  Reason: tried: '/opt/homebrew/opt/mbedtls/lib/libmbedcrypto.12.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/mbedtls/lib/libmbedcrypto.12.dylib' (no such file), '/opt/homebrew/opt/mbedtls/lib/libmbedcrypto.12.dylib' (no such file), '/usr/local/lib/libmbedcrypto.12.dylib' (no such file), '/usr/lib/libmbedcrypto.12.dylib' (no such file, not in dyld cache), '/opt/homebrew/Cellar/mbedtls/3.5.0/lib/libmbedcrypto.12.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/mbedtls/3.5.0/lib/libmbedcrypto.12.dylib' (no such file), '/opt/homebrew/Cellar/mbedtls/3.5.0/lib/libmbedcrypto.12.dylib' (no such file), '/usr/local/lib/libmbedcrypto.12.dylib' (no such file), '/usr/lib/libmbedcrypto.12.dylib' (no such file, not in dyld cache)
Abort trap: 6

**Comment by ZhongRuoyu** [comment]
Noticed that your `libidn2` is at 2.3.4 while the latest is 2.3.4_1. Please try `brew update && brew upgrade libidn2`.

**Reply by akaase** [reply]
Thanks for your help, you got me on the right mental track of the solution.

**Comment by akaase** [comment]
***@***.***:~$ brew update && brew upgrade libidn2
Updated 1 tap (homebrew/cask).
==> New Casks
xsplit-vcam
==> Outdated Formulae
bdw-gc            cmocka            gnu-getopt        imagemagick       libheif           libomp            librist           ***@***.***       ***@***.***       shared-mime-info  zeromq
cmake             gmp               guile             libevent          libidn2           libpthread-stubs  openjpeg          protobuf          rclone            x264

You have 21 outdated formulae installed.
You can upgrade them with brew upgrade
or list them with brew outdated.
==> Upgrading 1 outdated package:
libidn2 2.3.4 -> 2.3.4_1
==> Downloading https://ghcr.io/v2/homebrew/core/libidn2/manifests/2.3.4_1-1
################################################################################################################################################################################################# 100.0%
==> Fetching libidn2
==> Downloading https://ghcr.io/v2/homebrew/core/libidn2/blobs/sha256:d79edd6e4fa829e8e00f874b362068c67f8f9b4e76e58c8140e5b087ca924553
################################################################################################################################################################################################# 100.0%
==> Upgrading libidn2
  2.3.4 -> 2.3.4_1

==> Pouring libidn2--2.3.4_1.arm64_sonoma.bottle.1.tar.gz
🍺  /opt/homebrew/Cellar/libidn2/2.3.4_1: 79 files, 1MB
==> Running `brew cleanup libidn2`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
Removing: /opt/homebrew/Cellar/libidn2/2.3.3... (78 files, 1MB)
Removing: /opt/homebrew/Cellar/libidn2/2.3.4... (79 files, 1.3MB)
***@***.***:~$ ffmpeg
dyld[19870]: Library not loaded: /opt/homebrew/opt/mbedtls/lib/libmbedcrypto.12.dylib
  Referenced from: <9AB6FD1A-FE12-309C-8F22-AE6259869D59> /opt/homebrew/Cellar/librist/0.2.7_1/lib/librist.4.dylib
  Reason: tried: '/opt/homebrew/opt/mbedtls/lib/libmbedcrypto.12.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/mbedtls/lib/libmbedcrypto.12.dylib' (no such file), '/opt/homebrew/opt/mbedtls/lib/libmbedcrypto.12.dylib' (no such file), '/usr/local/lib/libmbedcrypto.12.dylib' (no such file), '/usr/lib/libmbedcrypto.12.dylib' (no such file, not in dyld cache), '/opt/homebrew/Cellar/mbedtls/3.5.0/lib/libmbedcrypto.12.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Cellar/mbedtls/3.5.0/lib/libmbedcrypto.12.dylib' (no such file), '/opt/homebrew/Cellar/mbedtls/3.5.0/lib/libmbedcrypto.12.dylib' (no such file), '/usr/local/lib/libmbedcrypto.12.dylib' (no such file), '/usr/lib/libmbedcrypto.12.dylib' (no such file, not in dyld cache)
Abort trap: 6
________________________________

**Comment by akaase** [comment, marked as answer]
You know what? I just did a full `brew upgrade` and that fixed it. ffmpeg runs fine now.
